File: backend-go/internal/domain/interfaces.go
package domain

import "github.com/shopspring/decimal"

type Breakpoint struct {
	LowerBound decimal.Decimal
	Fee        decimal.Decimal
}

type FeeRepository interface {
	GetBreakpoints(term int) []Breakpoint
}

type FeeInterpolationStrategy interface {
	Calculate(amount decimal.Decimal, lower Breakpoint, upper Breakpoint) (decimal.Decimal, error)
}

File: backend-dotnet/LoanFeeCalculator.Api/Domain/Interfaces.cs
using System.Collections.Generic;

namespace LoanFeeCalculator.Api.Domain;

public record Breakpoint(decimal LowerBound, decimal Fee);

public interface ILoanFeeRepository
{
    List<Breakpoint> GetBreakpoints(int term);
}

public interface IFeeInterpolationStrategy
{
    decimal Calculate(decimal amount, Breakpoint lower, Breakpoint upper);
}
